services/claim-backend/src/modules/claim/claim.service.ts

```diff
@@ -79,27 +85,131 @@ export class ClaimService {
 
     if (batch.type === BatchTypes.COMMUNITY_ACTIVITY) {
       const kyc = await this.getKycStatus();
-      await this.pendingRewardRepository.save(
-        await this.getPendingRewardsFor(
-          batch,
-          await this.getCommunityRewards(),
-          kyc,
-          'community',
-        ),
-      );
-
-      await this.pendingRewardRepository.save(
-        await this.getPendingRewardsFor(
+      const rewards = [
+        ...(await this.getCommunityRewards()),
+        ...(await this.getDelegationRewards()),
+      ];
+      for (const reward of rewards) {
+        const address = this.toChecksumAddress(reward.address);
+
+        let total = new BigNumber(0);
+        let claimable = new BigNumber(0);
+        let communityTotal = new BigNumber(0);
+        let delegationTotal = new BigNumber(0);
+
+        let isValid = true;
+        let invalidReason = null;
+
+        let pendingReward = await this.pendingRewardRepository.findOne({
+          address,
           batch,
-          await this.getDelegationRewards(),
-          kyc,
-          'delegation',
-        ),
-      );
+        });
+
+        if (pendingReward) {
+          total = total.plus(new BigNumber(pendingReward.total));
+          communityTotal = communityTotal.plus(
+            new BigNumber(pendingReward.communityTotal),
+          );
+          delegationTotal = delegationTotal.plus(
+            new BigNumber(pendingReward.delegationTotal),
+          );
+        } else {
+          pendingReward = new PendingRewardEntity();
+          pendingReward.batch = batch;
+          pendingReward.address = address;
+        }
+
+        total = total.plus(this.floatToBn(reward.total));
+        claimable = claimable.plus(total);
+
+        if (reward.type === RewardType.COMMUNITY) {
+          communityTotal = communityTotal.plus(this.floatToBn(reward.total));
+        }
+
+        if (reward.type === RewardType.DELEGATION) {
+          delegationTotal = delegationTotal.plus(this.floatToBn(reward.total));
```

Comment:
Doesn't `ethers.BigNumber` treat floats too?  I guess if u implemented it then it doesn't but we should add a PR to them with this, definitely.